src/Npgsql/NpgsqlCommand.cs

```diff
@@ -662,16 +679,19 @@ public void Unprepare()
             Log.Debug("Closing command's prepared statements", connector.Id);
             using (connector.StartUserAction())
             {
-                var sendTask = SendClose(connector, false);
+                var sendTask = SendClose(connector, async);
                 foreach (var statement in _statements)
                     if (statement.PreparedStatement?.State == PreparedState.BeingUnprepared)
                     {
-                        Expect<CloseCompletedMessage>(connector.ReadMessage(), connector);
+                        Expect<CloseCompletedMessage>(async ? await connector.ReadMessage(true) : connector.ReadMessage(), connector);
```

Comment:
Wouldn't that kinda destroy the purpose of making it async? If I understood correctly, wouldn't this mean that it would look something like `connector.ReadMessage(true).GetAwaiter().GetResult()`? Like if `Unprepare(true)` is called wouldn't that would mean, that the `ReadMessage` would be waited synchronously for even if you awaited the `Unprepare` method?

Reply:
Oh, I am just supposed to call `await connector.ReadMessage(async)`, I see.

Reply:
Yep